### PR TITLE
Add an API that tells if row key is supported or not

### PR DESCRIPTION
**Pull request by weiqingy**
## What changes were proposed in this pull request?

Add an API that tells if row key is supported or not.

## How was this patch tested?

Pass the current unit tests.

### CODE REVIEW

**Comment by weiqingy** [comment]
Hi, @dongjoon-hyun @bikassaha Could you please help to review this PR? Thanks.

File: core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/types/SHCDataType.scala [hunk 1]
```diff
@@ -28,6 +28,10 @@ trait SHCDataType extends Serializable {
   // Convert input to Byte Array (HBaseType)
   def toBytes(input: Any): Array[Byte]
 
+  // If lexicographic sort order is maintained, then return true.
+  // If return false, the data type can not be the table coder.
+  def isRowKeySupported(): Boolean = false
```
Comment (dongjoon-hyun) [inline core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/types/SHCDataType.scala]: So, `AvroType` simply use this method?

**Comment by dongjoon-hyun** [comment]
+1, LGTM.

File: core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/types/SHCDataType.scala (hunk 1 above)
Comment (weiqingy) [inline core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/types/SHCDataType.scala]: Yes.

**Comment by weiqingy** [comment]
Thanks for the review, @dongjoon-hyun .

**Pull request merged**